### ISSUE TITLE
[50] iBOT: Image BERT Pre-Training with Online Tokenizer

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#78]
[paper](https://arxiv.org/pdf/2111.07832.pdf)
[code](https://github.com/bytedance/ibot)

![image](https://user-images.githubusercontent.com/16400591/147716567-5fa0e26a-3438-4e24-bbce-dccb3f847e0f.png)

Online Tokenizer 를 이용한 ViT pretraining.
mask 를 self-distillation 하는 아이디어 제안.
자세한 건 아래에서.

# IBOT
BEiT(#37) 처럼 blockwise masking 을 우선 수행함.
2개의 view u, v 에 대해서 동일 적용
아래 그림을 보면서 이야기 해보자
![image](https://user-images.githubusercontent.com/16400591/147719298-404702a1-ef20-4bde-ba5c-038a0cf3f6b7.png)

teacher network 는 student 의 EMA 로 업데이트 됨.
student는 blockwise mask 가 되지만, teacher 에는 mask 없이 모두 들어감.
CLS token 과 patch 들을 distillation 함

projection head (3layer MLP) 는 weight-share 되어짐.
output dim 은 beit 처럼 8192.

# Result
## ImageNet
![image](https://user-images.githubusercontent.com/16400591/147719901-5858ae8d-5eb7-45cd-a752-ba9ffa1b0859.png)

에폭수를 감안하면 fair comparison 은 아니지만 이젠 이런 건 감안하려 한다.
그래도 에폭수 기재해준 게 어디냐..

## Detection & Segmentation 
![image](https://user-images.githubusercontent.com/16400591/147720283-db193bd3-5dd7-4d68-8b0c-587287413e1f.png)

## Ablations
sh 는 shared projection head.
cls token distillation 없으면 성능 확 떨어짐
~MIM without dVAE 를 사용하면서 (SimMIM, MAE, MaskFeat) 처럼 decoder도 안 붙이면 학습이 안 되는게 당연한 듯~
![image](https://user-images.githubusercontent.com/16400591/147720045-a763312a-9917-4fc8-b99f-a522808d6db9.png)
